### PR TITLE
Add CONTRIBUTING.md, closes #17

### LINKED ISSUE #17: Add CONTRIBUTING.md
A `CONTRIBUTING.md` file should be provided stating how potential contributors can help with the project.

<sub> Denis Neuling <denis.neuling@daimler.com>, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md)

### PR DESCRIPTION
**Pull request by denisneuling**
This PR adds a brief CONTRIBUTING.md file stating how potential contributors can help evolving this project.

The CONTRIBUTING.md file currently lacks information regarding code formatting/style. The IDSA Dataspace Connector project uses a slightly amended [sun_checks.xml](https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/sun_checks.xml) and [PMD](https://github.com/International-Data-Spaces-Association/DataspaceConnector/blob/main/scripts/ci/pmd-rules.xml) configuration which IMO could also be applied to this project as well. This is going to be added by a subsequent issue / pull request. 

<sub> Denis Neuling <denis.neuling@daimler.com>, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+### Discuss
+
+If you have questions or suggestions feel free to contact the project developers via
+the [project's "dev" list](https://dev.eclipse.org/mailman/listinfo/dataspaceconnector-dev). If you suppose there is a
+bug or issue contribute to discussions in
+[existing issues](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues?q=is%3Aissue+is%3Aopen)
+otherwise [file a new issue](#file-an-issue).
+
+### File an issue
```
Comment (DominikPinsel) [inline CONTRIBUTING.md:48]: is this a typo? "Fill an issue"

Comment (denisneuling) [reply CONTRIBUTING.md]: Has been reworded everywhere to "Create an Issue".

Comment (DominikPinsel) [reply CONTRIBUTING.md]: Looks like my English was just really bad. File an issue is just a term I didn't know. In case you want to revert your commit

Comment (denisneuling) [reply CONTRIBUTING.md]: All good, let's keep this phrase as it is right now - This way it's way clearer for the reader.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
+Contributing to Dataspace Connector
+===================================
+
+Thank you for your interest contributing to
+the [Eclipse Dataspace Connector](https://projects.eclipse.org/projects/technology.dataspaceconnector)!
+
+## Table of Contents
```
Comment (MoritzKeppler) [inline CONTRIBUTING.md:7]: let's check the length of the md in sum. Maybe toc is not needed at all.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+1. Log in to the [Eclipse foundation website](https://accounts.eclipse.org/user/login/). You will need to create an
+   account with the Eclipse Foundation if you have not already done so.
+2. Click on "Eclipse ECA", and complete the form.
+
+Be sure to use the same email address in your Eclipse account that you intend to use when you commit to GitHub.
+
+## How to Contribute
+
+### Discuss
```
Comment (MoritzKeppler) [inline CONTRIBUTING.md:40]: Move mailing list to "contact us", don't expect that to be the normal way of interacting. Better stay in GH.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+If you have questions or suggestions feel free to contact the project developers via
+the [project's "dev" list](https://dev.eclipse.org/mailman/listinfo/dataspaceconnector-dev). If you suppose there is a
+bug or issue contribute to discussions in
+[existing issues](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues?q=is%3Aissue+is%3Aopen)
+otherwise [create a new issue](#create-an-issue).
+
+### Create an Issue
+
+Although your contributions are always welcome please remember to search
```
Comment (MoritzKeppler) [inline CONTRIBUTING.md:50]: Important to mention that people should search for existing issues before opening a new one but I wouldn't start with that disclaimer. First of all we embrace every idea + contribution.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+If none of the existing issues meet your expectation feel free to file a new one at our projects
+corresponding [Github Issues page](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues/new).
+
+We
+use [Github's default label set](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels)
+to classify issues.
+
+### Submit a Pull Request
+
+* If you have not previously done so, please sign
```
Comment (MoritzKeppler) [inline CONTRIBUTING.md:63]: details are already given above

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+  the [Eclipse Contributor Agreement (ECA)](http://www.eclipse.org/legal/ecafaq.php).
+
+* Always apply the following copyright header to specific files in your work, attach the following boilerplate
```
Comment (MoritzKeppler) [inline CONTRIBUTING.md:66]: what about just linking to project handbook?
Guess most contributors will just copy&paste from existing files

**Review by MoritzKeppler — CHANGES_REQUESTED** [review]
thx for creating an first draft! 
added some suggestions as comments.

File: CONTRIBUTING.md (hunk 2 above)
Comment (denisneuling) [reply CONTRIBUTING.md:7]: I personally would like to keep the TOC since it provides orientation to the reader.

File: CONTRIBUTING.md (hunk 4 above)
Comment (denisneuling) [reply CONTRIBUTING.md]: You're right - wording was perhaps a bit harsh. Did some slight rewording. Please let me know whether it's better now.

File: CONTRIBUTING.md (hunk 5 above)
Comment (denisneuling) [reply CONTRIBUTING.md]: Yeah, that sentence was removed since it introduced redundancy.

File: CONTRIBUTING.md (hunk 6 above)
Comment (denisneuling) [reply CONTRIBUTING.md:64]: Added a link to the Eclipse Project Handbook to the preface of that section. Anyway I think licensing of source files is a conscious act and therefore we should keep this point here to provide clear guidance

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+    ```text
+    Copyright (c) {year} {owner}[ and others]
+
+    This program and the accompanying materials are made available under the
+    terms of the Apache License, Version 2.0 which is available at
+    https://www.apache.org/licenses/LICENSE-2.0
+
+    SPDX-License-Identifier: Apache-2.0
+
+    Contributors: 1
```
Comment (MoritzKeppler) [inline CONTRIBUTING.md:77]:
```suggestion
    Contributors: 
```

Comment (denisneuling) [reply CONTRIBUTING.md]: Removed.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+   account with the Eclipse Foundation if you have not already done so.
+2. Click on "Eclipse ECA", and complete the form.
+
+Be sure to use the same email address in your Eclipse Account that you intend to use when you commit to GitHub.
+
+## How to Contribute
+
+### Discuss
+
+If you feel there is a bug or issue contribute to discussions in
```
Comment (alexandrudanciu) [inline CONTRIBUTING.md:41]:
```suggestion
If you feel there is a bug or issue, contribute to discussions in
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+[existing issues](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues?q=is%3Aissue+is%3Aopen)
+otherwise [create a new issue](#create-an-issue).
+
+### Create an Issue
+
+If you have identified a bug or just want to share an idea to further enhance
+the project feel free to create a new issue at our projects corresponding
```
Comment (alexandrudanciu) [inline CONTRIBUTING.md:48]:
```suggestion
the project, feel free to create a new issue at our projects corresponding 
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+
+### Create an Issue
+
+If you have identified a bug or just want to share an idea to further enhance
+the project feel free to create a new issue at our projects corresponding 
+[Github Issues page](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues/new).
+
+Before doing so please consider searching for potentially suitable 
+[existing issues](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues?q=is%3Aissue+is%3Aopen). 
+There might already have been someone who already had the same idea or issue. Your input there is always valuable.
```
Comment (alexandrudanciu) [inline CONTRIBUTING.md:53]:
```suggestion
There might have been someone who already had the same idea or issue. Your input there is always valuable.
```

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+
+We also
+use [Github's default label set](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels)
+to classify issues and improve findability.
+
+### Submit a Pull Request
+
+In addition to the contribution guideline made available in the [Eclipse project handbook](https://www.eclipse.org/projects/handbook/#contributing)
+we would appreciate if your Pull Request applies to the following points:
```
Comment (alexandrudanciu) [inline CONTRIBUTING.md:62]:
```suggestion
we would appreciate if your Pull Request meets the following requirements:
```

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+
+* Always apply the following copyright header to specific files in your work, attach the following boilerplate
```
Comment (alexandrudanciu) [inline CONTRIBUTING.md:64]:
```suggestion
* Always apply the following copyright header to specific files in your work and include the following boilerplate
```

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+  Use the [imperative mood](https://github.com/git/git/blob/master/Documentation/SubmittingPatches)
+  as in "Fix bug" or "Add feature" rather than "Fixed bug" or "Added feature" and
+  [mention the GitHub issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
+  e.g. `transfer process: improve logging, closes #3`.
+
+  All committers, and all commits, are bound to
+  the [Developer Certificate of Origin.](https://www.eclipse.org/legal/DCO.php)
+  As such, all parties involved in a contribution must have valid ECAs and commits must
+  include [valid "Signed-off-by" entries](https://wiki.eclipse.org/Development_Resources/Contributing_via_Git). Commits
+  can be signed off by included the `-s` attribute in your commit message, for
```
Comment (alexandrudanciu) [inline CONTRIBUTING.md:95]:
```suggestion
  can be signed off by including the `-s` attribute in the commit command, for
```

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+* We prefer small changes and short living branches. Please do not combine independent things in one pull request.
+
+* Prefix the branch you're submitting with an issue number, e.g. `725-add-new-feature`.
+
+* Excessive branching and merging can make git history confusing. With that in mind please
+
+    * [Squash your commits](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_squashing)
+      down to a few commits, or one commit, before submitting a pull request and
+    * [Rebase](https://git-scm.com/book/en/v2/Git-Branching-Rebasing) your pull request changes on top of the current
+      master. Pull requests shouldn't include merge commits.
```
Comment (alexandrudanciu) [inline CONTRIBUTING.md:107]:
```suggestion
      main branch. Pull requests shouldn't include merge commits.
```

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+* Submit your pull request when ready. Make sure automatically kicked off checks pass, thus as
+
+    * _Intellectual Property Validation_ verifying the [Eclipse CLA](#eclipse-contributor-agreement) has been signed as
+      well as commits have been signed-off and
+    * _Continuous Integration_ performing various test proving functional integrity and alignment with our coding
+      conventions.
+
+### Contact Us
+
+If you have questions or suggestions do not hesitate to contact the project developers via
```
Comment (alexandrudanciu) [inline CONTRIBUTING.md:122]:
```suggestion
If you have questions or suggestions, do not hesitate to contact the project developers via
```

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+the [project's "dev" list](https://dev.eclipse.org/mailman/listinfo/dataspaceconnector-dev).
```
Comment (alexandrudanciu) [inline CONTRIBUTING.md:123]:
```suggestion
the [mailing list](https://dev.eclipse.org/mailman/listinfo/dataspaceconnector-dev). 
```

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+    * [Squash your commits](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_squashing)
+      down to a few commits, or one commit, before submitting a pull request and
+    * [Rebase](https://git-scm.com/book/en/v2/Git-Branching-Rebasing) your pull request changes on top of the current
+      master. Pull requests shouldn't include merge commits.
+
+* Add meaningful tests to verify your submission acts as expected.
+
+* Where code is not self-explanatory add documentation providing extra clarification.
+
+* Submit your pull request when ready. Make sure automatically kicked off checks pass, thus as
```
Comment (alexandrudanciu) [inline CONTRIBUTING.md:113]:
```suggestion
* Submit your pull request when ready. Make sure that the following types of automated tests run successfully:
```

**Review by alexandrudanciu — APPROVED** [review]
Suggested improvements were implemented.

**Review by mspiekermann — APPROVED** [review]

File: CONTRIBUTING.md (hunk 8 above)
Comment (MoritzKeppler) [inline CONTRIBUTING.md:41]:
```suggestion
If you feel there is a bug or an issue contribute to discussions in
```

File: CONTRIBUTING.md (hunk 10 above)
Comment (MoritzKeppler) [inline CONTRIBUTING.md:53]:
```suggestion
```

File: CONTRIBUTING.md (hunk 11 above)
Comment (MoritzKeppler) [inline CONTRIBUTING.md:62]:
```suggestion
we would appreciate if your pull request applies to the following points:
```

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -0,0 +1,123 @@
...
+use [Github's default label set](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels)
+to classify issues and improve findability.
+
+### Submit a Pull Request
+
+In addition to the contribution guideline made available in the [Eclipse project handbook](https://www.eclipse.org/projects/handbook/#contributing)
+we would appreciate if your Pull Request applies to the following points:
+
+* Always apply the following copyright header to specific files in your work, attach the following boilerplate
+  declaration, replacing the fields enclosed by curly brackets "{}" with your own identifying information. (Don't
```
Comment (MoritzKeppler) [inline CONTRIBUTING.md:65]:
```suggestion
* Always apply the following copyright header to specific files in your work replacing the fields enclosed by curly brackets "{}" with your own identifying information. (Don't
```

File: CONTRIBUTING.md [hunk 19]
```diff
@@ -0,0 +1,123 @@
...
+  All committers, and all commits, are bound to
+  the [Developer Certificate of Origin.](https://www.eclipse.org/legal/DCO.php)
+  As such, all parties involved in a contribution must have valid ECAs and commits must
+  include [valid "Signed-off-by" entries](https://wiki.eclipse.org/Development_Resources/Contributing_via_Git). Commits
+  can be signed off by included the `-s` attribute in your commit message, for
+  example, `git commit -s -am 'Interesting Commit Message.`
+
+* We prefer small changes and short living branches. Please do not combine independent things in one pull request.
+
+* Prefix the branch you're submitting with an issue number, e.g. `725-add-new-feature`.
```
Comment (MoritzKeppler) [inline CONTRIBUTING.md:100]:
```suggestion
```

File: CONTRIBUTING.md [hunk 20]
```diff
@@ -0,0 +1,123 @@
...
+
+* Add meaningful tests to verify your submission acts as expected.
+
+* Where code is not self-explanatory add documentation providing extra clarification.
+
+* Submit your pull request when ready. Make sure automatically kicked off checks pass, thus as
+
+    * _Intellectual Property Validation_ verifying the [Eclipse CLA](#eclipse-contributor-agreement) has been signed as
+      well as commits have been signed-off and
+    * _Continuous Integration_ performing various test proving functional integrity and alignment with our coding
```
Comment (MoritzKeppler) [inline CONTRIBUTING.md:117]:
```suggestion
    * _Continuous Integration_ performing various test
```

**Review by MoritzKeppler — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 21]
```diff
@@ -0,0 +1,120 @@
...
+  [mention the GitHub issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
+  e.g. `transfer process: improve logging, closes #3`.
+
+  All committers, and all commits, are bound to
+  the [Developer Certificate of Origin.](https://www.eclipse.org/legal/DCO.php)
+  As such, all parties involved in a contribution must have valid ECAs and commits must
+  include [valid "Signed-off-by" entries](https://wiki.eclipse.org/Development_Resources/Contributing_via_Git). Commits
+  can be signed off by included the `-s` attribute in your commit message, for
+  example, `git commit -s -am 'Interesting Commit Message.`
+
```
Comment (MoritzKeppler) [inline CONTRIBUTING.md:95]: signed-off is optional, see https://www.eclipse.org/lists/eclipse.org-committers/msg01291.html

```suggestion
  As such, all parties involved in a contribution must have valid ECAs. Additionally commits can
  include a ["Signed-off-by" entry](https://wiki.eclipse.org/Development_Resources/Contributing_via_Git). 
```

**Review by MoritzKeppler — APPROVED** [review]

**Pull request merged**